precice_config_graph/nodes.py

```diff
@@ -99,10 +145,67 @@ def __init__(self, participant: ParticipantNode, data: DataNode, mesh: MeshNode)
 
 
 class ExchangeNode:
-    def __init__(self, coupling_scheme: CouplingNode, data: DataNode, mesh: MeshNode, from_participant: ParticipantNode,
+    def __init__(self, coupling_scheme: CouplingNode|MultiCouplingNode, data: DataNode, mesh: MeshNode, from_participant: ParticipantNode,
                  to_participant: ParticipantNode):
         self.coupling_scheme = coupling_scheme
         self.data = data
         self.mesh = mesh
         self.from_participant = from_participant
         self.to_participant = to_participant
+
+
+class MultiCouplingNode:
+    def __init__(self, control_participant: ParticipantNode, participants: list[ParticipantNode],
+                 exchanges: list[ExchangeNode] = None):
+        # TODO control participant as first entry of participants[] ?
+        self.control_participant = control_participant
+        self.participants = participants
+        if exchanges is None:
+            self.exchanges = []
+        else:
+            self.exchanges = exchanges
+
+
+# TODO is a node for logging (https://precice.org/configuration-logging.html) needed? For me it does not make sense,
+#  as it is not used/connectd to anything else
+
+
+# TODO is export a “valid” node? Maybe good for info, but as stated in precice: "great feature for debugging", so not
+#  really a good way to use coupling data
+class ExportNode:
+    def __init__(self, participant: ParticipantNode):
+        self.participant = participant
+
+
+class ActionNode:
+    def __init__(self, name: str, participant: ParticipantNode, mesh: MeshNode, target_data: DataNode,
+                 timing: TimingType):
+        self.name = name
+        self.participant = participant
+        if mesh is None:
+            self.mesh = []
+        else:
+            self.mesh = mesh
```

Comment:
```suggestion
        self.mesh = mesh
```
this cumbersome workaround is only needed for lists!

Reply:
oh